csharp/ql/src/semmle/code/csharp/AnnotatedType.qll

```diff
@@ -77,24 +77,32 @@ private module Annotations {
   }
 
   newtype TAnnotations =
-    TAnnotationFlags(int flags) {
-      flags = getElementTypeFlags(_) or
-      flags = getTypeArgumentFlags(_, _) or
-      flags = getTypeParameterFlags(_, _)
+    TAnnotationFlags(int flags, Nullability n) {
+      exists(Element e | flags = getElementTypeFlags(e) and n = getElementNullability(e))
+      or
+      flags = 0 // n is unbound
     }
 
-  /** A set of annotations on a type. */
   class TypeAnnotations extends TAnnotations {
     int flags;
+    Nullability nullability;
 
-    TypeAnnotations() { this = TAnnotationFlags(flags) }
+    TypeAnnotations() { this = TAnnotationFlags(flags, nullability) }
 
-    /** Gets an annotation in this set of annotations. */
-    TypeAnnotation getAnAnnotation() { isSet(result.getBit()) }
+    /** Gets the flags (as a bitset) of this type annotation. */
+    int getFlags() { result = flags }
```

Comment:
I think making the change to the IPA type suggested above means we can remove this predicate.